tests/run-pass/fs.rs

```diff
@@ -182,3 +192,30 @@ fn test_rename() {
 
     remove_file(&path2).unwrap();
 }
+
+fn test_directory() {
+    let dir_path = prepare_dir("miri_test_fs_dir");
+    // Creating a directory should succeed.
+    create_dir(&dir_path).unwrap();
+    // Test that the metadata of a directory is correct.
+    assert!(dir_path.metadata().unwrap().is_dir());
```

Comment:
Could you also test the error that arises when trying to create this already existing directory again?